mtcpclient/calculate_test.go

```diff
@@ -35,23 +36,27 @@ func TestCalculateStdDev(t *testing.T) {
 		},
 	}
 	for _, test := range stdDevScenariosChecks {
-		var gc GroupOfConnections = []tcpclient.Connection{}
+
+		var gc *GroupOfConnections
+		gc = newGroupOfConnections(0)
 
 		var sum int
 		for i, connectionDuration := range test.durationsInSecs {
-			gc = append(gc, tcpclient.NewConnection(i, tcpclient.ConnectionEstablished,
+
+			gc.connections = append(gc.connections, tcpclient.NewConnection(i, tcpclient.ConnectionEstablished,
```

Comment:
if this brings the fix I think it has, i'd also try to test for a connection which status is closed (previously it was not considered)

Reply:
👍